andela/boondocks-bn-backend
# #169257416 Delete comment posted within the comment thread

#### What does this PR do?
Deletes comment posted within the comment thread 
#### Description of Task to be completed?
- Create a Comment model and comments table with following fields: 
`comment`, `userId`, `requestId`, `isVisible`
- Make table 1:N relations between `comments` and `requests` table and `comments` and `users` table
- Create `deleteComment` method in the `Comment` controller file and `deleteCommentById` in `CommentService`
#### How should this be manually tested?
- clone the repo to the local machine
- create a `trip` and create a comment on that trip, and get the `id` of the comment from the response after creation.
- in postman with method `PATCH` on `/api/v1/comment/:commentId/delete` where `:commentId` is the `id` gotten from previous step, click `send` button
In the postman body response is displayed the delete comment object.
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#169257416](https://www.pivotaltracker.com/story/show/169257416)
#### Screenshots (if appropriate)
1. Create trip
![image](https://user-images.githubusercontent.com/3137192/70059262-2628e900-15e9-11ea-8700-500120fe6883.png)

2.  Create a comment
![image](https://user-images.githubusercontent.com/3137192/70059430-72742900-15e9-11ea-89a9-f3edf0df314b.png)

3. Comments displaying on trip request
![image](https://user-images.githubusercontent.com/3137192/70059546-a94a3f00-15e9-11ea-8065-fe87826de5ab.png)

4. Deleting a comment
![image](https://user-images.githubusercontent.com/3137192/70059682-e0205500-15e9-11ea-87a8-70c739065036.png)

#### Questions:
N/A
